src/tsoder_bot.erl

```diff
@@ -14,13 +14,22 @@ init(State) ->
 terminate(Reason, StateName, StateData) ->
     error_logger:info_report([{reason, Reason}]).
 
-listen({message, Message}, Data) ->
+listen({message, Message}, Channel) ->
     error_logger:info_report([{message, Message}]),
-    {next_state, listen, Data};
+
+    option:foreach(
+      fun ({_, []}) -> Channel ! {message, "Hello there!"};
+          ({_, Name}) -> Channel ! {message, "Hello " ++ Name ++ "!"}
```

Comment:
Apparently it's a pain in the ass in Erlang https://noss.github.io/2009/03/29/erlang-io-format-to-string.html